DataProcessingTools/objects.py

```diff
@@ -73,18 +73,28 @@ def __init__(self, *args, **kwargs):
             redoLevel = kwargs.get("redoLevel", 0)
             saveLevel = kwargs.get("saveLevel", 0)
             fname = self.get_filename()
+            print('file name expected (new version): ', fname)
             verbose = kwargs.get("verbose", 1)
             if redoLevel == 0 and os.path.isfile(fname):
                 self.load(fname)
                 if verbose > 0:
                     print("Object loaded from file {0}".format(fname))
             else:
-                # create object
-                self.create(*args, **kwargs)
-                if self.dirs and saveLevel > 0:
-                    self.save()
+                # attempt old hashing - os/version dependent
+                fname = self.get_filename(legacy=1)
+                print('file name expected (old version): ', fname)
+                if redoLevel == 0 and os.path.isfile(fname):
+                    self.load(fname)
                     if verbose > 0:
-                        print("Object saved to file {0}".format(fname))
+                        print("Object loaded from file {0}".format(fname))                
```

Comment:
by default, we now get the hash of the arguments using hashlib purely (without mixing in hickle). if it is not able to find, before proceeding to create a new object it tries to get another hash value, using the old hickle method.

Reply:
object will always be saved with new hash